pandas/core/series.py

```diff
@@ -3437,6 +3438,16 @@ def _reduce(self, op, name, axis=0, skipna=True, numeric_only=None,
         # dispatch to ExtensionArray interface
         if isinstance(delegate, ExtensionArray):
             return delegate._reduce(name, skipna=skipna, **kwds)
+        if (isinstance(delegate, ABCDatetimeIndex) and
+                name in ['mean', 'median', 'std', 'min', 'max']):
+            if numeric_only or filter_type:
+                raise TypeError
+            method = getattr(delegate, name)
+            try:
+                return method(skipna=skipna, **kwds)
+            except TypeError:
+                # kludge because not all reduction implementations take skipna
```

Comment:
That's just because we haven't implemented DatetimeArray / TimedeltaArray, correct? So this block is being removed once that's done?

Reply:
This is because the reductions defined in DatetimeIndexOpsMIxin (min, max, argmin, argmax) don't have a skipna kwarg

Reply:
Any thoughts on if we should/can just make the index versions conform to everything else?

Reply:
To me, supporting `skipna` on the index classes seems easier than trying to work around it in a shared implementation. Would you agree?

Reply:
I agree, will make PR to adapt the Index reduction signatures to match the others.  @jorisvandenbossche @jreback you on board with this plan?

Reply:
Yes, supporting skipna in the Index versions as well seems fine to me.

Reply:
Changed the signatures for the relevant Index reductions, haven't actually _implemented_ the changes, much less tests.  Before jumping into that: @jreback on board with this plan?